src/main/java/org/zalando/fauxpas/FauxPas.java

```diff
@@ -56,4 +59,44 @@
         return predicate;
     }
 
+    public static <R> Function<Throwable, R> partially(final ThrowingFunction<Throwable, R, Throwable> function) {
+        return throwable -> {
+            if (throwable instanceof CompletionException) {
+                final CompletionException original = (CompletionException) throwable;
+                final Throwable cause = original.getCause();
+
+                try {
+                    return function.tryApply(cause);
+                } catch (final CompletionException e) {
+                    if (e.getCause() == cause) {
+                        throw original;
+                    } else {
+                        throw e;
+                    }
+                } catch (final RuntimeException e) {
+                    if (e == cause) {
+                        throw original;
```

Comment:
So here you throw original CompletionException in case of runtime exception where in the else branch you rethrow RuntimeException as is.